Readme update (#496)
microsoft/vscode-azurearmtools: README.md

[![Version](https://vsmarketplacebadge.apphb.com/version/msazurermtools.azurerm-vscode-tools.svg)](https://marketplace.visualstudio.com/items?itemName=msazurermtools.azurerm-vscode-tools) [![Installs](https://vsmarketplacebadge.apphb.com/installs-short/msazurermtools.azurerm-vscode-tools.svg)](https://marketplace.visualstudio.com/items?itemName=msazurermtools.azurerm-vscode-tools) [![Build Status](https://dev.azure.com/ms-azuretools/AzCode/_apis/build/status/vscode-azurearmtools)](https://dev.azure.com/ms-azuretools/AzCode/_build/latest?definitionId=10)

The Azure Resource Manager (ARM) Tools for Visual Studio Code provides language support, resource snippets, and resource auto-completion to help you create and validate Azure Resource Manager templates.

![ARM Tools creating web app](./images/arm-tools.gif)

## Language Server

- Azure Resource Manager template language server for providing ARM template language completion, validation, and error guidance.

> - Resources are validated against Azure schemas which provides validation against resource type and apiVersion properties.

Reviewer: how about "based on resource type and apiVersion"?

Author: Like this?

Resources are validated against Azure schemas which provides validation based on resource type and apiVersion.

Reviewer: Yes, but grammar nit: Ambiguous what "which" refers to, making it unclear whether it should be "provide" or "provides".  How about a comma after "schemas"?

- ARM Template Outline view for easy navigation through large templates
- Colorization for Template Language Expressions (TLE)
- Analyze and validate JSON syntax, JSON schema conformance, and string expressions.

## Intellisense

>   - [Built-in function names](https://go.microsoft.com/fwlink/?LinkID=733958)

Reviewer: Built-in functions (I know, not yours)

Author: Fixed

  - [Parameter references](https://go.microsoft.com/fwlink/?LinkID=733959)
  - [Variable references](https://go.microsoft.com/fwlink/?LinkID=733960)
  - [resourceGroup() properties](https://azure.microsoft.com/en-us/documentation/articles/resource-group-template-functions/#resourcegroup)
  - [subscription() properties](https://azure.microsoft.com/en-us/documentation/articles/resource-group-template-functions/#subscription)
  - Properties of references to variables that are objects

> ## Other features:

Reviewer: Rename is mentioned twice below

Author: Fixed.

## License

> The source code in our [public repository](https://github.com/Microsoft/vscode-azurearmtools) is licensed under the [MIT license](LICENSE.md). The public source code currently contains functionality related to the parsing and validation of template expression strings but does not contain functionality related to JSON parsing and validation or backend template validation.

Reviewer: ```suggestion
The source code in our [public repository](https://github.com/Microsoft/vscode-azurearmtools) is licensed under the [MIT license](LICENSE.md) and may be locally built and used in accordance with this license. The public source code currently contains functionality related to the parsing and validation of template expression strings but does not contain functionality related to JSON parsing and validation or backend template validation, as these are subject to a different license.
```

Reviewer: Does these changes make more sense?

Author: They do, however what if we simplify it down a bit.

```
## License

The source code in our [public repository](https://github.com/Microsoft/vscode-azurearmtools) is licensed under the [MIT license](LICENSE.md) and may be locally built and used in accordance with this license.

When the extension is [published](https://marketplace.visualstudio.com/items?itemName=msazurermtools.azurerm-vscode-tools) to the VS Code Marketplace, it is bundled with the Azure Resource Manager language service binaries. The extension as bundled and published this way is licensed under a traditional Microsoft product license.
```

> The extension as it is built-in Azure DevOps and [published](https://marketplace.visualstudio.com/items?itemName=msazurermtools.azurerm-vscode-tools) to the VS Code Marketplace is a distribution of the public repository and is bundled with the Azure Resource Manager language service binaries. The published extension and language service binaries are licensed under a traditional Microsoft product license.

Reviewer: the hyphen is wrong - "the extension as it is built" "in Azure DevOps..."

Guess it's awkward, maybe you can rephrase?

Author: I've made some updates, do these work?

Reviewer: Perfect

> The extension as it is built-in Azure DevOps and [published](https://marketplace.visualstudio.com/items?itemName=msazurermtools.azurerm-vscode-tools) to the VS Code Marketplace is a distribution of the public repository and is bundled with the Azure Resource Manager language service binaries. The published extension and language service binaries are licensed under a traditional Microsoft product license.

Reviewer: ```suggestion
When the extension is [published](https://marketplace.visualstudio.com/items?itemName=msazurermtools.azurerm-vscode-tools) to the VS Code Marketplace, it is bundled with the Azure Resource Manager language service binaries. The extension as bundled and published this way is licensed under a traditional Microsoft product license.
```
